src/main/java/at/fifthwheel/battleship/GameSetupController.java

```diff
@@ -0,0 +1,420 @@
+package at.fifthwheel.battleship;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.geometry.BoundingBox;
+import javafx.geometry.Bounds;
+import javafx.scene.Node;
+import javafx.scene.layout.*;
+import javafx.scene.paint.Color;
+import javafx.scene.shape.Rectangle;
+import javafx.scene.control.Button;
+
+import java.awt.*;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class GameSetupController {
+
+    private final int GRID_SIZE = 10;
+    private final int CELL_SIZE = 35;
+    public Button btnRotateShip_P1;
+    public Button btnRotateShip_P2;
+    public Button btnContinue;
+
+    @FXML
+    private Pane shipContainer_P1;
+    @FXML
+    private Pane shipContainer_P2;
+
+    @FXML
+    private GridPane gameSetupGrid_P1;
+    @FXML
+    private GridPane gameSetupGrid_P2;
+
+    private ObservableList<Ship> shipsP1;
+    private ObservableList<Ship> shipsP2;
+
+    private Map<Rectangle, Ship> rectangleShipMap_P1;
+    private Map<Rectangle, Ship> rectangleShipMap_P2;
+
+    private final int shipsMidIndex = 2;
+    private final double shipsColumnOffset = 20.0 + CELL_SIZE;
+    private final double shipsRowOffset = 20.0;
+
+    private Map<Rectangle, Point> shipOrigins_P1;
+    private Map<Rectangle, Point> shipOrigins_P2;
+
+    private Rectangle lastPlacedShipRect;
+
+    private boolean p1Done = false;
+
+    @FXML
+    public void initialize() {
+
+        shipsP1 = FXCollections.observableArrayList(
+                new Ship(5, 1),
+                new Ship(4, 1),
+                new Ship(3, 1),
+                new Ship(3, 1),
+                new Ship(2, 1)
+        );
+        shipsP2 = FXCollections.observableArrayList(
+                new Ship(5, 1),
+                new Ship(4, 1),
+                new Ship(3, 1),
+                new Ship(3, 1),
+                new Ship(2, 1)
+        );
+
+        rectangleShipMap_P1 = new HashMap<>();
+        rectangleShipMap_P2 = new HashMap<>();
+
+        shipOrigins_P1 = new HashMap<>();
+        shipOrigins_P2 = new HashMap<>();
+
+        initialize(true);
```

Comment:
Eine initializePlayer1 und eine initializePlayer2 Methode und duplizierung rausziehen, dieser boolean is sehr verwirrend